### PR TITLE
Site Editor: Improve the header animation

### PR DESCRIPTION
**Pull request by youknowriad**
Follow-up to #60363 
Also based on #58924

## What?

This PR moves the editor header into the frame (which is more logical IMO) and improve the animation.

Result:

https://github.com/WordPress/gutenberg/assets/272444/ece4db6e-e28b-4dc6-a741-30219d303c76

### CODE REVIEW

**Comment by jasmussen** [comment]
Looks good, and I know this is something that's been advocated for.

But it's also something we tried in the past, and then specifically moved it _out_ of the frame, in order to better orchestrate animation elsewhere. Are those motivations no longer valid?

**Comment by youknowriad** [comment]
>  in order to better orchestrate animation elsewhere

Can you clarify this? I think at some point we just wanted the animation to come from the top but I don't feel like that's necessary or that it brings any value. For me it just adds complexity.

**Comment by jasmussen** [comment]
Mainly about keeping the frame to be _content only_. I don't have a strong opinion, and happy to move forward, animation works.

**Comment by jameskoster** [comment]
I think it's hard to answer that question without a comprehensive prototype that considers all the variables. 

That said, this feels like an improvement on trunk to me. Mostly because there's only a single element animating as a whole, rather than two separate elements which feels a bit disjointed, particularly in the side-by-side data view layout. 

I still think it would be a good idea to prototype the ideal, as the animation details can also be fine-tuned, but that doesn't need to hold up this PR.

**Comment by jasmussen** [comment]
Just to visualize, the inspector causes a jump both in trunk and in this branch. This branch:

site editor

Trunk:

trunk

That's not necessarily a blocker, but it does speak to the larger orchestration challenge. For example, will the inspector need to live outside of this frame still, as far as the unification work with data views goes? If yes, then this PR might not be the right approach after all.

**Comment by youknowriad** [comment]
I think this PR is important in more than just the animation, let me explain.

 - Leaving the header within the "layout" of the site editor means that the "header" is something that is "global" to all pages of the site editor. It's like saying there's a "header" area that the different pages/routes can inject UI in. 
 - Moving the header inside the "frame" means that it's just part of one of the two existing areas that routes/pages can inject UI in. (content and frame).

It is clear that the current header is not something "global" to all the pages of the site editor, it's clear that it's specific to "editor" pages. 
It is also clear that today we don't have any other pages (other than editor pages) that inject another kind of "header" in that particular area.

Meaning that I believe the second approach (this PR) is more suited for our use-cases and is de facto simpler: It doesn't require formalizing a new "header" area as it seems that the only pages that need it today are the ones that are "editors" and they can just inject it in the "frame" area.

**Comment by youknowriad** [comment]
> That's not necessarily a blocker, but it does speak to the larger orchestration challenge. For example, will the inspector need to live outside of this frame still, as far as the unification work with data views goes? If yes, then this PR might not be the right approach after all.

I don't really think the inspector or other "editor specific" panels should leave outside of the frame. Anything that moves outside of the frame forces us create a new "routable area" in the site editor. An area that pages can decide which UI to inject in It's easy to see that this doesn't scale very well, especially since most of these are editor specific UI.

**Comment by jameskoster** [comment]
@youknowriad that makes sense to me, though it's worth connecting the dot about potentially invoking the inspector from data views for quick/bulk edits. Does that affect anything in the logic outlined above?

quick-edit

**Comment by youknowriad** [comment]
So what should we do here? Can I get an approval maybe?

**Comment by jameskoster** [comment]
This aligns with the [prototype](https://www.figma.com/proto/e3u7Q1ChpEbl485EvSoD58/Frame-animation?page-id=20%3A3702&type=design&node-id=20-4138&viewport=11%2C-1151%2C0.28&t=I4XzjODr5GAksmW2-1&scaling=min-zoom&starting-point-node-id=20%3A4212&show-proto-sidebar=1&mode=design) we've been exploring so I think it's worth trying. 

I updated the animation profile to match the prototype. It would be nice to apply this to the frame as well, but I couldn't work out how to do that.

**Review by jameskoster — APPROVED** [review]

**Comment by SaxonF** [comment]
+1 on this being an improvement and simplification in terms of moving parts and technically too. I see the preview frame as site in view that can transition to edit mode, so header inside makes sense to me.

**Comment by youknowriad** [comment]
@jameskoster The frame animation uses React sprint, it is possible to use "easing" for React sprint animations but I don't understand how to translate the config you provided for framer motion to the config required for "easings" in React Spring https://react-spring.dev/docs/advanced/config#easings

For now, I just adapted the duration of the animation.

**Comment by stokesman** [comment]
I ❤️ this change. Yet a `git bisect` led me here after I noticed the Document Bar animation (when switching between a page/template) is not working in trunk. I've tried working out a solution and but haven’t found anything that seems appropriate. In that process I also discovered that in “Distraction free” mode the header hover reveal animation is also broke after this change. Some points that may aid investigation:
-  In the Post editor both animations work which is a bit surprising considering the next point. (Aside: the DocumentBar animation doesn’t run the first time switching to template but I only mention it to set the expectation, it seems irrelevant otherwise).
- When certain props of `EditorProvider` change all children have their local state reset. This too happens in the Post editor but there seems to be less renders and somehow that’s allowing the animations to work.

One thing I tried to fix it was moving `EditorProvider` to only wrap the `content` of `InterfaceSkeleton` and it works but I didn’t run all e2e tests so not sure it wouldn’t cause other problems.

I hope this is helpful and I'm very interested to see what a good solution looks like. Oh, in case anyone would like to verify here’s the commit from this PR 3ece636 and the parent commit (where issues mentioned aren’t present) a7a62f7.

**Comment by youknowriad** [comment]
The distraction free animation is already tracked and I've been working on a potential fix here #61579 (which I believe is the right fix, but I wonder if it's the right time for it).

I'm not sure whether it impacts the other one too but it might be good to track it as well.

**Comment by stokesman** [comment]
I've added an issue for the Document Bar animation ☝️. I tested #61579 for it and saw no impact. I didn’t rebase the branch but doubt that would have made a difference.

I also went ahead and made #61874 to test the fix I found. No tests failed but that doesn’t mean I trust it. Riad, if you can can have look maybe you'd have insight of whether or not it’s a bad idea.

**Pull request merged**
